src/stan/io/dump.hpp

```diff
@@ -242,8 +241,8 @@ class dump_reader {
     scan_optional_long();
     size_t d = 0;
     try {
-      d = boost::lexical_cast<size_t>(buf_);
-    } catch (const boost::bad_lexical_cast& exc) {
+      d = std::stoull(buf_);
+    } catch (const std::exception& exc) {
```

Comment:
If we are using the `std::sto*` functions they can only call two exceptions

```
> [std::invalid_argument](https://en.cppreference.com/w/cpp/error/invalid_argument) if no conversion could be performed.
> [std::out_of_range](https://en.cppreference.com/w/cpp/error/out_of_range) if the converted value would fall out of the range of the result type or if the underlying function (std::strtol or std::strtoll) sets [errno](https://en.cppreference.com/w/cpp/error/errno) to [ERANGE](https://en.cppreference.com/w/cpp/error/errno_macros).
```

I'd rather have try-catches for each of these

Reply:
https://en.cppreference.com/w/cpp/string/basic_string/stol

Reply:
noted - backed out changes.